lang/c/src/schema.c

```diff
@@ -546,16 +549,63 @@ avro_schema_enum_number_of_symbols(const avro_schema_t enum_schema)
 	return enump->symbols->num_entries;
 }
 
-int
-avro_schema_record_field_append(const avro_schema_t record_schema,
-				const char *field_name,
-				const avro_schema_t field_schema)
+static int
+is_equal_schema_type(const avro_schema_t field_schema, const json_t *field_default)
+{
+	int  json_type = json_typeof(field_default);
+
+	switch (avro_typeof(field_schema)) {
+	case AVRO_STRING:
+	case AVRO_BYTES:
+		return json_type == JSON_STRING;
+
+	case AVRO_INT32:
+	case AVRO_INT64:
+		return json_type == JSON_INTEGER;
+
+	case AVRO_FLOAT:
+	case AVRO_DOUBLE:
+		return json_type == JSON_REAL;
+
+	case AVRO_BOOLEAN:
+		return (json_type == JSON_TRUE) || (json_type == JSON_FALSE);
+
+	case AVRO_NULL:
+		return json_type == JSON_NULL;
+
+	case AVRO_ARRAY:
+		return json_type == JSON_ARRAY;
+
+	case AVRO_RECORD:
+	case AVRO_ENUM:
```

Comment:
As I understand, default for `enum` is just a string: 
```
avro type | json type | example
-- | -- | --
null | null | null
boolean | boolean | true
int,long | integer | 1
float,double | number | 1.1
bytes | string | "\u00FF"
string | string | "foo"
record | object | {"a": 1}
enum | string | "FOO"
array | array | [1]
map | object | {"a": 1}
fixed
```
from http://avro.apache.org/docs/1.7.7/spec.html#Schema+Resolution